core/azure-core/src/main/java/com/azure/core/polling/Poller.java

```diff
@@ -0,0 +1,215 @@
+        if (cancelInitiated
+            || (pollResponse != null && pollResponse.status() != PollResponse.OperationStatus.IN_PROGRESS)) {
+            return;
+        }
+        //Time to call cancel
+        cancelOperation.accept(this);
+        cancelInitiated = true;
+    }
+
+    /**
+     * This will poll and send PollResponse. If you had stopped polling erlier, we will enable polling again.
+     *
+     * @return Return poll response as Flux
+     **/
+    public Flux<PollResponse<T>> poll() {
+        setStopPolling(false);
+        if (fluxDisposable != null && !fluxDisposable.isDisposed()) {
+            fluxDisposable.dispose();
+        }
+        return fluxHandle;
+    }
+
+    /**
+     * This will block till poll operation is complete
+     *
+     * @return returns poll response
+     **/
+    public PollResponse<T> block() {
+        return poll().blockLast();
+    }
+
+    /**
+     * Get whether or not this PollStrategy's long running operation is done.
+     *
+     * @return Whether or not this PollStrategy's long running operation is done.
+     */
+    private Mono<PollResponse<T>> sendPollRequestWithDelay() {
+        return Mono.defer(() -> delayAsync().then(Mono.defer(() -> {
+            if (!isPollingStopped() && !isTerminalState()) {
+                pollResponse = pollOperation.apply(pollResponse);
+            } else if (!isTerminalState()) {
+                return Mono.empty();
+            }
+            return Mono.just(pollResponse);
+        })));
+    }
+
+    /*Helper function*/
+    private boolean isTerminalState() {
+        return pollResponse != null && (
+            pollResponse.status() == PollResponse.OperationStatus.SUCCESSFULLY_COMPLETED
+                || pollResponse.status() == PollResponse.OperationStatus.FAILED
+                || pollResponse.status() == PollResponse.OperationStatus.USER_CANCELLED);
+
+    }
+
+    /**
+     * If this PollerOptions has a pollIntervalInMillis value, return an Mono that is delayed by the
+     * number of seconds that are in the pollIntervalInMillis value. If this PollerOptions doesn't have
+     * a pollIntervalInMillis value, then return an Single with no delay.
+     *
+     * @return A Mono with delay if this PollerOptions has a pollIntervalInMillis value.
+     */
+    private Mono<Void> delayAsync() {
+        Mono<Void> result = Mono.empty();
+        if (this.pollerOptions.getPollIntervalInMillis() > 0) {
+            result = result.delaySubscription(Duration.ofMillis(
+                (long) (this.pollerOptions.getPollIntervalInMillis() * this.pollerOptions.getPollIntervalGrowthFactor())
+            ));
+        }
+        return result;
+    }
+
+    /**
+     * This will stop polling
+     **/
+    public void stopPolling() {
+        setStopPolling(true);
+    }
+
+    /**
+     * Enable auto polling
+     **/
+    public void enablePolling() {
+        setStopPolling(false);
+    }
```

Comment:
I think we need to be more clear about what 'stopPolling' and 'enablePolling' mean. This really means to stop / start auto-polling. I would probably remove the stopPolling and enablePolling method, and just have the setStopPolling method renamed to 'setAutoPollingEnabled()', and 'isPollingStopped()' would be 'isAutoPollingEnabled()'